server/src/main/java/org/elasticsearch/index/translog/TranslogReader.java

```diff
@@ -77,75 +66,8 @@
      */
     public static TranslogReader open(
             final FileChannel channel, final Path path, final Checkpoint checkpoint, final String translogUUID) throws IOException {
-
-        try {
-            InputStreamStreamInput headerStream = new InputStreamStreamInput(java.nio.channels.Channels.newInputStream(channel),
-                channel.size()); // don't close
-            // Lucene's CodecUtil writes a magic number of 0x3FD76C17 with the
```

Comment:
this PR removes this `LUCENE_CODEC_HEADER_BYTE` check etc. since we don't support it, correct. Maybe we should keep it and fail with a descriptive error message instead?

Reply:
I've restored this.